crates/red_knot_python_semantic/resources/mdtest/assignment/augmented.md

```diff
@@ -45,3 +45,107 @@ x -= 1
 # TODO: should error, once operand type check is implemented
 reveal_type(x)  # revealed: int
 ```
+
+## Method union
+
+```py
+def bool_instance() -> bool:
+    return True
+
+flag = bool_instance()
+
+class Foo:
+    if bool_instance():
+        def __iadd__(self, other: int) -> str:
+            return "Hello, world!"
+    else:
+        def __iadd__(self, other: int) -> int:
+            return 42
+
+f = Foo()
+f += 12
+
+reveal_type(f)  # revealed: str | int
+```
+
+## Partially bound `__iadd__`
+
+```py
+def bool_instance() -> bool:
+    return True
+
+class Foo:
+    if bool_instance():
+        def __iadd__(self, other: str) -> int:
+            return 42
+
+f = Foo()
+f += "Hello, world!"
+
+reveal_type(f)  # revealed: int
+```
+
+## Partially-bound with `__add__`
+
+```py
+def bool_instance() -> bool:
+    return True
+
+class Foo:
+    def __add__(self, other: int) -> str:
```

Comment:
I think `other: int` is wrong for what you're intending to test here, since it would actually mean that the binary fallback should also fail and `str` would not be possible.
```suggestion
    def __add__(self, other: str) -> str:
```

Reply:
Sorry, yes, you're right. (Wouldn't be enforced today IIUC but yes agree.)